subprojects/common/graph/src/main/kotlin/com/avito/graph/Operation.kt

```diff
@@ -0,0 +1,14 @@
+package com.avito.graph
+
+interface Operation {
+    val id: String
+    val duration: Double
+
+    /**
+     * ids of predecessor operations
+     */
+    val predecessors: Set<String>
+}
+
+fun Operation.isLeaf() =
```

Comment:
On the other hand, we have the opposite case. For now, they are named as "roots".
To avoid confusion, I'll use the correct original terms for DAG - sink and source.

Reply:
Fixed in https://github.com/avito-tech/avito-android/pull/985/commits/f2038794bedf48b0e4f02583c5ad2f4826547df7